### PR TITLE
Enforce version, binable changes linting

### LINKED ISSUE #7999: Enforce version-change linting
CI currently does not enforce version changes.

### PR DESCRIPTION
**Pull request by psteckler**
In buildkite, remove the `soft-fail` flag for `binable` and `version` change-detection linting, so that changes are errors. Remove the temporary, duplicate version-change detection in `OCaml.dhall`.

(Emacs `dhall-mode` changed the formatting, sorry.)

Closes #7999.

### CODE REVIEW

File: buildkite/src/Jobs/Lint/Fast.dhall [hunk 1]
```diff
@@ -1,66 +1,76 @@
 let Prelude = ../../External/Prelude.dhall
+
 let B = ../../External/Buildkite.dhall
 
 let B/SoftFail = B.definitions/commandStep/properties/soft_fail/Type
```
Comment (0x0I) [inline buildkite/src/Jobs/Lint/Fast.dhall:5]: Should be ok to remove with the soft-fails

File: buildkite/src/Jobs/Lint/Fast.dhall [hunk 2]
```diff
@@ -1,66 +1,76 @@
...
-                          (["CI=true", "BASE_BRANCH_NAME=$BUILDKITE_PULL_REQUEST_BASE_BRANCH" ])
-                          ("./scripts/compare_ci_diff_binables.sh")
-          , label = "Optional fast lint steps; binable compatability changes"
-          , key = "lint-optional-binable"
-          , target = Size.Medium
-          , soft_fail = Some (B/SoftFail.Boolean True)
-          , docker = None Docker.Type
+in  Pipeline.build
+      Pipeline.Config::{
+      , spec = JobSpec::{
```
Comment (0x0I) [inline buildkite/src/Jobs/Lint/Fast.dhall:32]: rogue comma

File: buildkite/src/Jobs/Lint/Fast.dhall [hunk 3]
```diff
@@ -1,66 +1,76 @@
...
+        , dirtyWhen = [ S.strictlyStart (S.contains "src/") ]
+        , path = "Lint"
+        , name = "Fast"
         }
-    ]
-  }
+      , steps =
+        [ Command.build
+            Command.Config::{
+            , commands
```
Comment (0x0I) [inline buildkite/src/Jobs/Lint/Fast.dhall:40]: `commands = ...`

File: buildkite/src/Jobs/Lint/Fast.dhall (hunk 2 above)
Comment (psteckler) [reply buildkite/src/Jobs/Lint/Fast.dhall]: inserted by Emacs, let me try again

**Review by 0x0I — APPROVED** [review]

**Comment by imeckler** [comment]
!approved-for-mainnet
